OloEngine/src/OloEngine/Particle/ParticleSystem.cpp

```
@@ -506,6 +522,60 @@ namespace OloEngine
+                gp.InitialColor = m_Pool.m_InitialColors[idx];
+                gp.InitialVelocitySize = glm::vec4(m_Pool.m_InitialVelocities[idx], m_Pool.m_Sizes[idx]);
+                gp.Misc = glm::vec4(m_Pool.m_InitialSizes[idx], m_Pool.m_Rotations[idx], 1.0f, 0.0f);
+            }
+            m_GPUSystem->EmitParticles(gpuParticles);
+
+            // Remove the CPU-side particles (GPU owns them now)
+            m_Pool.Resize(m_Pool.GetMaxParticles());
+        }
+
+        // Fill simulation params from current module settings
+        GPUSimParams params;
```

Clang (14.0.6)

[warning] variable 'params' is not initialized

(cppcoreguidelines-init-variables)